main.py

```diff
@@ -81,10 +50,23 @@ def create_images_with_text_and_bounding_box(n, word_list, width, height):
 
     for _ in range(0, n):
         image = Image.new("RGB", (width, height))
-
-        rand_color = (randrange(255), randrange(255), randrange(255))
-        bounding_boxes.append(render_random_words_at_random_position_and_return_aabb(
-            image, word_list, text_color=rand_color))
+        image_aabb = AABB((0, 0), (width, height))
+        text_count = randrange(min_text_count, max_text_count)
+        texts = []
+        for i in range(text_count):
+            rand_color = (randrange(255), randrange(255), randrange(255))
+            text = None
+            while True:
+                x = randrange(width + 1)
+                y = randrange(height + 1)
+                text = generate_text(
+                    word_list, (x, y), text_color=rand_color)
+                if(not is_text_overlapping(text, texts, image_aabb)):
+                    break
+            texts.append(text)
+        for text in texts:
```

Comment:
I think this is okay, because, if something breaks while generating the images he did not save any file. With separating those iterations the images are saved when everything went fine.